Feature player

AinzZamma opened the pull request:
Open New Version 1.0.0

AinzZamma committed:
Basic player movement

Here, the basic movement of the player using WASD has been added, camera control via mouse, jump to SPACE and a scene where you can check everything

AinzZamma committed:
The basic movement of the player v 0.0.1

There is an updated version of the basic player movement here.

AinzZamma committed:
Basic player movement v 0.0.2

AinzZamma committed:
Basic player movement v 1.0.0

AinzZamma committed:
Basic player movement v 1.0.1

i-shindev commented on Assets/Features/Player/Scripts/BasePlayerLocomotion.cs:
```diff
@@ -0,0 +1,43 @@
+    
+    public abstract class BasePlayerLocomotion : MonoBehaviour
+    {
+        byte[] utf8 = Encoding.UTF8.GetBytes("���������");
```
.... зачем.


![image](https://github.com/user-attachments/assets/682a051e-3ed0-41b6-bffc-dec2e8c941f7)

AinzZamma replied:
Это тот самый один из способов конвертирования

i-shindev commented on Assets/Features/Player/Scripts/CameraLocomotionFeature.cs:
```diff
@@ -0,0 +1,31 @@
+namespace Feature.Player
+{
+    
+    //�������� ������
```
исправь комментарии, преобразовав файлы с кодом в UTF-8 формат (в блокноте сохрани как UTF-8)

i-shindev commented on Assets/Features/Player/Scripts/CharacterControllerLocomotion.cs:
```diff
@@ -0,0 +1,83 @@
+        public float jumpHeight = 1.5f;
+        public float gravity = -9.81f;
+
+        private Vector3 cameraRotation;
```
_cameraRotation поправильней будет

i-shindev commented on Assets/Features/Player/Scripts/CharacterControllerLocomotion.cs:
```diff
@@ -0,0 +1,83 @@
+
+            if (playerCamera == null)
+            {
+                playerCamera = Camera.main;
```
Так, вот это лучше правда не делать, поскольку если внезапно в сцене появится другая камера, я разочаруюсь в том, что игрок использует другую как главную

сделай поле в CharacterControllerLocomotion.cs
```c#
[SerializeField] private Transform _camera;
```
тогда никто не мешает сделать, чтобы камера уже имела определённый localPosition и localRotation в самом редакторе, а не скрипте.

i-shindev replied:
убери из BasePlayerLocomotion поле playerCamera

i-shindev commented on Assets/Features/Player/Scripts/CharacterControllerLocomotion.cs:
```diff
@@ -0,0 +1,83 @@
+            transform.rotation = Quaternion.Euler(0f, CameraRotation.y, 0f);
+
+
+
```
было бы хорошо, если бы код разделяла максимум 1 пустая строка

i-shindev reviewed (changes requested):
Чуть чуть осталось исправлений и можно будет сливать.

AinzZamma committed:
Basic player movement v 1.0.2

svvoooyyyy committed:
Small code and scene fixes

- moved playerCamera from BasePlayerLocomotion to CharacterControllerLocomotion
- deleted CapsuleCollider component from player because CapsuleCollider is already included in CharacterController component

svvoooyyyy committed:
folder remove

- samples folder deleted

i-shindev committed:
Little refactoring. All works. Merging.

TODO: Ceiling contact doesn't change velocity to zero, but if we'll use Rigidbody, that would be solved

i-shindev reviewed (approved):
Рефакторинг
